HelpSource/Classes/Dictionary.schelp

```diff
@@ -135,40 +139,41 @@ d.getPairs;
 ::
 
 method::associationAt
-Access the link::Classes/Association:: that has the given key.
+Access the link::Classes/Association:: that has the given key. Element is checked for equality (not identity).
 code::
-d = (robot: 99);
-d.associationAt(\robot);	// Get the value associated with key
+d = Dictionary["robot" -> 99];
+d.associationAt("robot");	// Get the value associated with key
 ::
 
 method::findKeyForValue
-Try to find a given value and return its key.
+Try to find a given value and return its key. Element is checked for equality (not identity).
 code::
-d = (hello: 1, whello: 1976);
+d = Dictionary.newFrom([\hello, 1, \whello, 77]);
 d.findKeyForValue(1);
 ::
 
 method::matchAt
 The dictionary's keys are used as conditions against which the arbitrary item is matched. See: link::Reference/matchItem::
+Returns the associated value or nil if no key is matching the item.
 note::
 if an item matches multiple criteria, the value returned is arbitrary. This is because a dictionary is an unordered collection. It's the user's responsibility to make sure that criteria are mutually exclusive.
 ::
 list::
-## If the key is an object, the item will be matched by identity (if key === item, the value will be returned).
+## If the key is an object, the item will be matched by equality (if key == item, the value will be returned). For identity matching see subclasses: link::Classes/IdentityDictionary:: or link::Classes/Event::.
```

Comment:
I've never used this functionality so I'm not sure what's the desired behavior, but it doesn't seem like equality matching actually works here:

```supercollider

(
d = Dictionary.newFrom([
    \abc: \alpha,
	"def": \beta,
]);
);

d.matchAt(\abc); // -> alpha
d.matchAt("def"); // -> nil
// although this works
d["def"]; // -> beta
```

Reply:
Oh, yes you are right. Checking back in my testing files I just realized, that I mixed that example up with associationAt.

The implementation of matchAt uses [matchItem](http://doc.sccode.org/Reference/matchItem.html#.matchItem), which is checking for identity. So I think it would be best to keep the original description `matched by identity (if key === item, the value will be returned).`

Reply:
Yes, if you just want to use the dictionary as a lookup, you'd not use `matchAt`, but `at`. I think `matchAt` and `matchItem` assume an `IdentityDictionary` for efficiency. I do think that `matchItem` for string should return true for equality.